docs/eom_base.rst

```diff
@@ -0,0 +1,372 @@
+==============================
+Two Mass Spring Damper Example
+==============================
+
+This example walks through the creation of a two mass spring damper system,
+manually entered to the `eombase` class and numerically simulated using
+`pydy.system.System`. ::
+
+    >>> import matplotlib.pyplot as plt
+    >>> from numpy import linspace
+    >>> from pydy.system import System
+    >>> from sympy import symbols, Matrix
+    >>> from sympy.physics.mechanics import dynamicsymbols, eombase
+
+The first step is to define the dynamic and constant symbols that will be used
+in the system. ::
+
+    >>> x1, x2, u1, u2 = dynamicsymbols('x1 x2 u1 u2')
+    >>> m1, c1, k1 = symbols('m1 c1 k1')
+    >>> m2, c2, k2 = symbols('m2 c2 k2')
+
+Next step is to define the equations of motion in multiple forms:
+
+[1] x' = F(x, t, r, p)
+
+[2] M(x, p) x' = F(x, t, r, p)
+
+[3] M(q, p) u' = F(q, u, t, r, p)
+    q' = G(q, u, t, r, p) ::
+
+    >>> mm = Matrix([[m1,  0],
+    ...             [0,  m2]])
+    >>> f = Matrix([-(k1+k2)*x1 + k2*x2 - (c1+c2)*u1 + c2*u2,
+    ...             k2*x1 - k2*x2 + c2*u1 - c2*u2])
+    >>> mm_full = Matrix([[1, 0,  0,  0],
+    ...                   [0, 1,  0,  0],
+    ...                   [0, 0, m1,  0],
+    ...                   [0, 0,  0, m2]])
+    >>> f_full = Matrix([u1,
+    ...                  u2,
+    ...                  -(k1+k2)*x1 + k2*x2 - (c1+c2)*u1 + c2*u2,
+    ...                  k2*x1 - k2*x2 + c2*u1 - c2*u2])
+    >>> G = Matrix([u1, u2])
+    >>> RHS = mm_full.LUsolve(f_full)
+
+Now the various forms of varible entry to `eombase` will be initiated ::
+
+    >>> coordinates = (x1, x2)
+    >>> speeds = (u1, u2)
+    >>> states = (x1, x2, u1, u2)
+
+This example will also show `system`'s ability to calculate an arbitrary user
+defined functions along with the states. Specifically the kinetic energy and
+potential energy  of the whole system is going to be determined. ::
+
+    >>> KE = 0.5 * (m1*u1**2 + m2*u2**2)
+    >>> PE = symbols("PE")
+    >>> out_eqns = {"kinetic energy": KE, PE: 0.5*k1*x1**2 + 0.5*k2*(x2-x1)**2}
+
+The `eombase` instances are now ready to be initialized. To showcase possible
+input combinations, all three of the equations of motion forms will be used. ::
+
+    >>> eom1 = eombase.EOM(coordinates, RHS, speeds=speeds, 
+    ...                    output_eqns=out_eqns)
+    >>> eom2 = eombase.EOM(states, f_full, mass_matrix=mm_full,
+    ...                    num_coordinates=2)
```

Comment:
I don't think `num_coordinates` is sufficient. For example, what if the states are in this order:

`q1, q2, u3, u2, q3, u1`

We should allow any order of the states, as users that manually generate them might want an arbitrary  order.

The other thing to note is that `q_i` does not necessarily correspond to `u_i` in the general sense and `len(q)!=len(u)`. So, there needs to be a way to tell the class which indices are coordinates or speeds.

Maybe `coord_idxs=(3, 2, 5, 9)`.